chiron/mcmc.py

```diff
@@ -112,103 +170,802 @@ def run(
+        self,
+        current_sampler_state: SamplerState,
+        current_thermodynamic_state: ThermodynamicState,
+        current_reduced_pot: float,
+        current_nbr_list: Optional[PairsBase] = None,
+    ) -> Tuple[SamplerState, ThermodynamicState, float, float, Optional[PairsBase]]:
         """
-        self._check_state_compatiblity(old_state, new_state)
-        old_system = self.system(old_state)
-        new_system = self.system(new_state)
+        Propose a new state and calculate the log proposal ratio.
+
+        This will accept the relevant quantities for the current state, returning the proposed state quantities
+        and the log proposal ratio.
+
+        This will need to be defined for each new move.
+
+        Parameters
+        ----------
+        current_sampler_state : SamplerState, required
+            Current sampler state.
+        current_thermodynamic_state : ThermodynamicState, required
+            Current thermodynamic state.
+        current_reduced_pot : float, required
+            Current reduced potential.
+        current_nbr_list : PairsBase, required
+            Neighbor list associated with the current state.
 
-        energy_before_state_change = old_system.compute_energy(old_state.position)
-        energy_after_state_change = new_system.compute_energy(new_state.position)
-        # Implement the logic to compute the acceptance probability
+        Returns
+        -------
+        proposed_sampler_state : SamplerState
+            Proposed sampler state.
+        proposed_thermodynamic_state : ThermodynamicState
+            Proposed thermodynamic state.
+        proposed_reduced_pot : float
+            Proposed reduced potential.
+        log_proposal_ratio : float
+            Log proposal ratio.
+        proposed_nbr_list : PairsBase
+            Proposed neighbor list. If not defined, this will be None.
+
+        """
         pass
 
-    def accept_or_reject(self, probability):
+    def _accept_or_reject(
+        self,
+        log_proposal_ratio,
+        key,
+        method,
+    ):
+        """
+        Accept or reject the proposed state with a given methodology.
+        """
+        # define the acceptance probability
+        if method == "metropolis":
+            import jax.random as jrandom
+
+            compare_to = jrandom.uniform(key)
+            if -log_proposal_ratio <= 0.0 or compare_to < jnp.exp(log_proposal_ratio):
+                return True
+            else:
+                return False
+
+
+class MetropolisDisplacementMove(MCMove):
+    def __init__(
+        self,
+        displacement_sigma=1.0 * unit.nanometer,
+        nr_of_moves: int = 100,
+        atom_subset: Optional[List[int]] = None,
+        report_frequency: int = 1,
+        reporter: Optional[LangevinDynamicsReporter] = None,
+        update_stepsize: bool = True,
+        update_stepsize_frequency: int = 100,
+    ):
+        """
+        Initialize the Displacement Move class.
```

Comment:
We can definitely improve this documentation to make sure it is clear.  

This will displace all particles each step (each particle by a unique random vector, simultaneously), rather than moving a single particle each step.  atom_subset is just the subset of atoms that will have the random displacements applied to it (not a means to move a group of atoms collectively).

Reply:
Documentation of these routines have been updated.